### PR TITLE
Accessibility Slides

### PR DESCRIPTION
**Pull request by neelghosh**

### CODE REVIEW

**Comment by neelghosh** [comment]
- [ ] What is Accessibility (1 minutes)
    - Definition of accessibility 
    - Accessibility vs Usability  
- [ ] Accessibility and the Law (2 minutes)
    - Description of legal obligations in Canada and abroad
- [ ] P.O.U.R. (10 minutes)
    - Description of the key principles of accessibility in development
- [ ] Implementing Accessibility - Starting with Design (2 minutes)
    - Description of common barriers to accessibility in design (to be overcome)
- [ ] Accessible Design (5 minutes)
    - Detailing how to achieve accessibility in design 
- [ ] Accessible Design (5 minutes)
    - Detailing how to achieve accessibility in design 
- [ ] Accessible Development (5 minutes)
    - Detailing how to achieve accessibility in development 
- [ ] Testing Accessibility (5 minutes)
    - Detailing how to achieve accessibility in design 

~ 25 minutes

File: slides/src/content/a11y.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+Accessibility is generally defined as giving people of all abilities opportunities to participate fully in everyday life.  This more recently includes use of digital technology.
+
+Accessibility can often be thought of as being complimentary to, or a subset of usability.  A feature on a site may be highly usable but inaccessible.  
+
+- For example, a button may be easy to find and click, but inaccessible via a keyboard
+- This prevents visually impaired users from browsing a site that otherwise has high usability
+
+---
+<!-- .slide: id="a11y" -->
+## Accessibility and the Law
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: 1. Combine this with the slide following
2. Cut the detail significantly - developers won't care much about specifics.

File: slides/src/content/a11y.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+- Use relative width and font sizing over absolutes (this may be obvious but...)
+- Use headers in sequnce (H1 - H6)
+
+---
+<!-- .slide: id="a11y" -->
+## Accessible Development (Continued)
+
+- To help navigation for visually impaired users, consider developing an offscreen-text CSS class to further mark up headers and indicate start and end of sections
+
+```ts
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: css not ts

File: slides/src/content/a11y.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+- Don't use the 'summary' attribute on the Table tag if you must create a layout table (or set the 'summary' attribute to null
+- Don't use TH or Caption tags on layout tables
+- Use relative width and font sizing over absolutes (this may be obvious but...)
+- Use headers in sequnce (H1 - H6)
+
+---
+<!-- .slide: id="a11y" -->
+## Accessible Development (Continued)
+
+- To help navigation for visually impaired users, consider developing an offscreen-text CSS class to further mark up headers and indicate start and end of sections
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: make stronger: not "consider developing" but "develop"

File: slides/src/content/a11y.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+	.offscreen-text {
+     	left: -3000px;
+     	position: absolute;
+     	width: 500px;
+     	font-size: 100%;}
+```
+- Mark up text with this class beside sections, including those ARIA tagged with `<aside>`, `<main>`, `<nav>`, `<footer>`, `<header>` to indicate where the user is
+- Examples:
+  - 'start of navigation'
+  - 'end of navigation'
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: great - now show me all of this in an Angular project

File: slides/src/content/a11y.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+
+---
+<!-- .slide: id="a11y" -->
+## Testing Accessibility 
+
+Testing accessibility involves testing the combination of design considerations and how they are implemented in code.
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: Remove this sentence.

File: slides/src/content/a11y.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+- Accessibility testing should typically start with manual testing then automated
+  - Develop appropriate scripts
+  - Automate using any number of tools
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: These three points are true of all kinds of testing - we should skip anything that's generic.

File: slides/src/content/a11y.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+- Testing against a checklist of items as compared to WCAG 2.0 AA standards can ensure compliance
+- Test for usability and functionality simultaneously and don't compromise these for accessibility
+
+
+---
+<!-- .slide: id="a11y" -->
+## Unit Testing for Accessiblity
+
+- Code can be ‘unit tested’ for accessibility (along with functional unit testing) before it is checked into the repository.
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: 1.  no quotes around unit tested
1.  show me the tests!!

File: slides/src/content/a11y.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+- Testing against a checklist of items as compared to WCAG 2.0 AA standards can ensure compliance
+- Test for usability and functionality simultaneously and don't compromise these for accessibility
+
+
+---
+<!-- .slide: id="a11y" -->
+## Unit Testing for Accessiblity
+
+- Code can be ‘unit tested’ for accessibility (along with functional unit testing) before it is checked into the repository. 
+  - Can be as simple as running a browser add-on or plug-in such as the Firefox Accessibility plug-in
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: 1. show me how to do this (with screenshots)

File: slides/src/content/a11y.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+  - Helps ensure a certain level of conformance to the selected accessibility standard. 
+  - Consistent with Agile development practices.
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: As a developer, I have no idea what this would mean in practice.

File: slides/src/content/a11y.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+---
+<!-- .slide: id="a11y" -->
+## Unit Testing for Accessiblity
+
+- Code can be ‘unit tested’ for accessibility (along with functional unit testing) before it is checked into the repository. 
+  - Can be as simple as running a browser add-on or plug-in such as the Firefox Accessibility plug-in
+  - Helps ensure a certain level of conformance to the selected accessibility standard. 
+  - Consistent with Agile development practices. 
+- Will not eliminate all potential accessibility defects:
+  - But will likely minimize the number of retroactive bug fixes that are required for defects not discovered until the testing phase.
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: Again, this is all high-level fuzziness: show me what to do.

File: slides/src/content/a11y.md [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+- Colour:  Ensure foreground:background ratios are 4:5:1 for normal text and 3:1 for large text
+- Images: Images must be alt tagged or have figures/numbers describing the image
+<br><br>![alt text](content/images/a11yfont.png "example of accessible font and colour ratio") ![alt text](content/images/a11ycolour2.png "example of accessible font and colour ratio")
+<br>Two images showing samples of a accessible font types and fore/back ground colour ratios
+<br>
+
+---
+<!-- .slide: id="a11y" -->
+## Accessible Design - Tables
+- Tables:  Do not use tables for layout, only data, and explicitly label columns and rows
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: label how?

File: slides/src/content/a11y.md [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+<br><br>![alt text](content/images/a11yinactab.png "example of inaccessible data table")
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: Blank lines will do paragraph breaks - no need for `<br>`

File: slides/src/content/a11y.md [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+- Using the above items, it is recommended a standard template for your web projects be developed
+- Demo:  Accessible forms-template in angular 2 
+
+---
+<!-- .slide: id="a11y" -->
+## Accessible Development
+
+To acheive accessibility set out in design, developers can use the following tips, do's, and don'ts.
+
+General Considerations
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: delete "To achieve..." lines and "General Considerations" line - these are slides, so it's OK to go straight into the bullet points.

File: slides/src/content/a11y.md [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+- Attempt to pin down client requirements for browsers to be supported before you start
+- Design and test accessibility includng unit testing from the start of a project instead of making it an add on later
+
+
+---
+<!-- .slide: id="a11y" -->
+## Accessible Development - HTML
+
+- Use the Title attribute on HTML anchor links to add more information about a link and/or its consequences
+  - `<a href=
```
Comment (gvwilson) [inline slides/src/content/a11y.md]: ???

**Pull request merged**